### ISSUE TITLE
Hyperparameter Optimization Setup

### ISSUE THREAD

**Issue by DCalhas** [DCalhas/eeg_to_fmri#5]
Use Bayesian Optimization Algorithm.

Parameters to have into account:
- Learning rate
- Output Dimension
- Number convolution layers BOLD Encoder (do this after having a generator of layers with consistent input/output) with a desired output
- Number convolution layers EEG Encoder (do this after having a generator of layers with consistent input/output) with a desired output
- Number deconvolution layers Shared Space Decoder (do this after having a generator of layers with consistent input/output) with a desired output
- Regularization - kernel and bias
- Linear Combination loss coefficients (a x contrastive loss + b x recontruction loss) for the Encoder
